plugins/bookshortcuts.koplugin/main.lua

```diff
@@ -104,37 +96,42 @@ function BookShortcuts:getSubMenuItems()
+                G_reader_settings:readSetting("BookShortcuts_directory_action", "FM") == "FM" and FM_text or last_text) end,
             keep_menu_open = true,
             sub_item_table = {
+                {
+                    text = FM_text,
+                    radio = true,
+                    checked_func = function() return G_reader_settings:readSetting("BookShortcuts_directory_action") == "FM" end,
+                    callback = function() G_reader_settings:saveSetting("BookShortcuts_directory_action", "FM") end,
+                },
                 {
                     text = last_text,
+                    radio = true,
                     checked_func = function() return G_reader_settings:readSetting("BookShortcuts_directory_action") == "Last" end,
                     callback = function() G_reader_settings:saveSetting("BookShortcuts_directory_action", "Last") end,
                 },
                 {
-                    text = FM_text,
-                    checked_func = function() return G_reader_settings:readSetting("BookShortcuts_directory_action") == "FM" end,
-                    callback = function() G_reader_settings:saveSetting("BookShortcuts_directory_action", "FM") end,
+                    text = _("Recursively search folders"),
+                    enabled_func = function() return G_reader_settings:readSetting("BookShortcuts_directory_action") == "Last" end,
+                    checked_func = function() return G_reader_settings:isTrue("BookShortcuts_recursive_directory") end,
+                    callback = function() G_reader_settings:flipNilOrFalse("BookShortcuts_recursive_directory") end,
                 },
             },
-        },
-        {
-            text = _("Recursively search folders"),
-            keep_menu_open = true,
-            checked_func = function() return G_reader_settings:isTrue("BookShortcuts_recursive_directory") end,
-            enabled_func = function() return G_reader_settings:readSetting("BookShortcuts_directory_action") == "Last" end,
-            callback = function() G_reader_settings:flipNilOrFalse("BookShortcuts_recursive_directory") end,
             separator = true,
-        }
+        },
     }
-    for k,v in FFIUtil.orderedPairs(self.shortcuts.data) do
+    for k in ffiUtil.orderedPairs(self.shortcuts.data) do
+        local mode = lfs.attributes(k, "mode")
+        local icon = mode and (mode == "file" and "\u{F016} " or "\u{F114} ") or "\u{F48E} "
+        local text = mode == "file" and k:gsub(".*/", "") or k
         table.insert(sub_item_table, {
-            text = k,
+            text = icon .. text,
             callback = function() self:onBookShortcut(k) end,
             hold_callback = function(touchmenu_instance)
                 UIManager:show(ConfirmBox:new{
-                    text = _("Do you want to delete this shortcut?"),
+                    text = _("Do you want to delete this shortcut?" .. "\n\n" .. k .. "\n"),
```

Comment:
Is the final newline intentional?

Reply:
For better readability 
https://github.com/koreader/koreader/blob/69d79cdc0e0bbfa7a04d3cfe90be6f902935c0c9/frontend/apps/filemanager/filemanagerutil.lua#L383

Reply:
That seems a bit different, but if it's on purpose that's fine by me. I'd overlooked this problem though. ;-)
```suggestion
                    text = _("Do you want to delete this shortcut?") .. "\n\n" .. k .. "\n",
```